api/services/Ban.js

```diff
@@ -93,13 +93,13 @@ exports.updateBanlist = async function (redToken, username, banlistEntry, friend
     if (knownAlt && lines[i].includes(knownAlt) || lines[i].includes(username)
         ||_.intersection(lines[i].match(/(\d{4}-){2}\d{4}/g), friend_codes).length) {
       // User was an alt account, modify the existing line instead of creating a new one
-      let blocks = lines[i].split(/\s*\|\s*/);
+      let blocks = lines[i].replace('\\_','_').split(/\s*\|\s*/);
       let user_regex = '(?:^|\\s)('+username+(knownAlt ? '|'+knownAlt : '')+')(?:,|$)';
       let fc_match = _.intersection(blocks[1].match(/(\d{4}-){2}\d{4}/g), friend_codes).length;
       if (blocks.length !== 5 || !(fc_match || blocks[0].match(new RegExp(user_regex)))) {
         continue;
       }
-      blocks[0] = _.union(blocks[0].match(/[\w-]{1,20}/g), [username]).join(', ');
+      blocks[0] = _.union(blocks[0].match(/[\w-]{1,20}/g), [username]).join(', ').replace('_','\\_');
```

Comment:
If we do stick to variables with pre-escaped underscores, we could alter this regex to be something like `/(?:[A-Za-z0-9]|\\_){1,20}/g`, but I'm wondering if we should just split `blocks[0]` into a secondary array on `/\s*,\s*/` instead.

_Ideally_ `blocks[0]` should contain nothing except for comma-separated Reddit usernames, but up until a few months ago, we did have a name entry that was in the format `IRC nick: irc_nick` in one of the banlist entries. As we've seen, unusual entries such as this can lead to unexpected results, so I'm thinking just treating whatever's separated by commas as a name might be best, without placing further restrictions on what else it has to be.